strategies.py

```diff
@@ -158,12 +158,13 @@ def play_move(self, c):
         self.comments.append(self.root.describe())
         try:
             self.root = self.root.maybe_add_child(coords.to_flat(c))
-        except go.IllegalMove:
+        except go.IllegalMove as e:
             dbg("Illegal move")
             if not self.two_player_mode:
                 self.searches_pi.pop()
             self.comments.pop()
-            return False
+            raise e
```

Comment:
`raise e` here will loose the callstack information. Use a plain `raise` to reraise the exception while preserving the callstack. You won't need the `as e` in the `except` line above either.